pkg/syncer/storagepool/util.go

```diff
@@ -118,43 +118,52 @@ func getHostMoIDToK8sNameMap(ctx context.Context) (map[string]string, error) {
 }
 
 // findAccessibleNodes returns the k8s node names of ESX hosts (limited to
-// clusterID) on which the given datastore is mounted and accessible. The
+// clusterIDs) on which the given datastore is mounted and accessible. The
 // values in the map tells whether the ESX host is in Maintenance Mode.
 func findAccessibleNodes(ctx context.Context, datastore *object.Datastore,
-	clusterID string, vcclient *vim25.Client) (map[string]bool, error) {
+	clusterIDs []string, vcclient *vim25.Client) (map[string]bool, error) {
```

Comment:
we do not support cross-zonal datastore, so for node which belongs to cluster which is at index 1 in clusterIDs, we will require unnecessary check in cluster at index 0.

Can we improve this?

Reply:
I've updated the code to keep track of which cluster a datastore belongs to and use that info wherever required.